core/src/main/java/com/netflix/iceberg/BaseTableScan.java

```diff
@@ -129,28 +137,53 @@ public TableScan select(Collection<String> columns) {
 
     // all of the filter columns are required
     requiredFieldIds.addAll(
-        Binder.boundReferences(table.schema().asStruct(), Collections.singletonList(rowFilter), true));
+        Binder.boundReferences(table.schema().asStruct(), Collections.singletonList(rowFilter), caseSensitive));
 
     // all of the projection columns are required
     requiredFieldIds.addAll(TypeUtil.getProjectedIds(table.schema().select(columns)));
 
     Schema projection = TypeUtil.select(table.schema(), requiredFieldIds);
 
-    return new BaseTableScan(ops, table, snapshotId, projection, rowFilter);
+    return new BaseTableScan(ops, table, snapshotId, projection, rowFilter, caseSensitive);
   }
 
   @Override
   public TableScan filter(Expression expr) {
-    return new BaseTableScan(ops, table, snapshotId, schema, Expressions.and(rowFilter, expr));
+    return new BaseTableScan(ops, table, snapshotId, schema, Expressions.and(rowFilter, expr), caseSensitive);
+  }
+
+  private final class InclusiveManifestEvaluatorCacheKey {
+    Integer specId;
+    Boolean caseSensitive;
+
+    InclusiveManifestEvaluatorCacheKey(Integer specId, Boolean caseSensitive) {
+      this.specId = specId;
+      this.caseSensitive = caseSensitive;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      InclusiveManifestEvaluatorCacheKey that = (InclusiveManifestEvaluatorCacheKey) o;
+      return specId.equals(that.specId) &&
+              caseSensitive.equals(that.caseSensitive);
+    }
+
+    @Override
+    public int hashCode() {
+      return java.util.Objects.hash(specId, caseSensitive);
+    }
+
   }
 
-  private final LoadingCache<Integer, InclusiveManifestEvaluator> EVAL_CACHE = CacheBuilder
+  private final LoadingCache<InclusiveManifestEvaluatorCacheKey, InclusiveManifestEvaluator> EVAL_CACHE = CacheBuilder
```

Comment:
The `EVAL_CACHE` name is wrong and misleading. (Sorry, my fault.)

Despite the name, this is an instance field, not a static field. As a result, the cache is not shared across scans and it is safe to use just the spec ID as the key because the `caseSensitive` argument never changes for a scan instance (it is `final`). That way, you can remove the cache key class and avoid making this change, other than to reference `caseSensitive`.

Reply:
I definitely missed that `private final`. Will fix.